src/Telemetry/AzureDevOpsClient/AzureDevOpsClient.cs

```diff
@@ -155,6 +162,40 @@ public async Task<WorkItem> CreateRcaWorkItem(string project, string title, Canc
             return JsonConvert.DeserializeObject<WorkItem>(json);
         }
 
+        public async Task<string> TryGetLogContents(string logUri, CancellationToken cancellationToken)
+        {
+            for (int i = 0; i < _retryNumber; i++)
+            {
+                using (var request = new HttpRequestMessage(HttpMethod.Get, logUri))
+                {
+                    request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("text/plain"));
+
+                    var response = await _httpClient.SendAsync(request, cancellationToken);
+
+                    if (response.StatusCode == HttpStatusCode.TooManyRequests)
+                    {
+                        if (response.Headers.TryGetValues("Retry-After", out var values) &&
+                            int.TryParse(values.First(), out var seconds))
+                        {
+                            int randomDelay;
+                            lock (_randomNumberGenerator)
+                            {
+                                randomDelay = _randomNumberGenerator.Next(_retryDelayMin, _retryDelayMax);
+                            }
+                            await Task.Delay((int)(TimeSpan.FromSeconds(seconds).TotalMilliseconds + randomDelay), cancellationToken);
+                            continue;
+                        }
+                    }
+
+                    response.EnsureSuccessStatusCode();
+
+                    return await response.Content.ReadAsStringAsync(cancellationToken);
+                }
+            }
+
+            return string.Empty;
```

Comment:
Should we signal with `null` maybe?

Also `EnsureSuccessStatusCode` throws so we could even throw here as well?